### PR TITLE
Di 24914   integration test   for 100k items 

### PR DESCRIPTION
**Pull request by NoamPepperi**

### CODE REVIEW

File: server-side/integration-tests/services/items-service.ts [hunk 1]
```diff
@@ -3,47 +3,88 @@ import { ItemRelations, itemsResourceObject } from 'shared';
...
         return items;
     }
 
+    // resource search return 1000 items per page, we need list of 4000 items for huge test so we need to get 4 pages
     async getUsersItems(): Promise<ItemRelations[]> {
+        let items: ItemRelations[] = [];
+        // for huge test we need the stock to be with 100000 items but the file only contain 4K enteries
+        // for small test we need 500 entieries
+        const itemsToAddCounter: number = this.numberOfItems === 100000 ? 4003 : this.numberOfItems;
+        while (items.length < itemsToAddCounter) {
```
Comment (simcha-waintrob) [inline server-side/integration-tests/services/items-service.ts:30]: page_size=-1 don't return all items?

Comment (NoamPepperi) [reply server-side/integration-tests/services/items-service.ts:31]: No

File: server-side/integration-tests/services/items-service.ts [hunk 2]
```diff
@@ -3,47 +3,88 @@ import { ItemRelations, itemsResourceObject } from 'shared';
...
-    async prepareUserItems(): Promise<ItemRelations[]> {
-        let items: ItemRelations[] = await this.getUsersItems()
+    async prepareUserItems(numberOfItems: number): Promise<ItemRelations[]> {
+        this.numberOfItems = numberOfItems;
+        let items: ItemRelations[] = await this.getUsersItems();
+        const itemsCount = await this.getItemsCount();
         // generating items if there are not enough items for the tests
-        if (items.length < this.NUMBER_OF_ITEMS) {
+        if (itemsCount < this.numberOfItems) {
             await this.createNewItems(items.length);
```
[the remark this answers is not in the document]
Comment (NoamPepperi) [reply server-side/integration-tests/services/items-service.ts:18]: I don't understand the question

File: server-side/integration-tests/services/items-service.ts [hunk 3]
```diff
@@ -1,49 +1,75 @@
...
-            PageSize: this.NUMBER_OF_ITEMS,
-        })).Objects as ItemRelations[];
+            PageSize: 1,
+            IncludeCount: true
+        })).Count as number;
     }
 
 
-    async createNewItems(itemsCounter) {
+    async createNewItems(itemsCounter: number) {
```
Comment (simcha-waintrob) [inline server-side/integration-tests/services/items-service.ts:50]: add comment please

Comment (simcha-waintrob) [reply server-side/integration-tests/services/items-service.ts:50]: and rename the function name to be clearer

File: server-side/integration-tests/test-commands/import-tests/import-file/huge-file-command.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+export class HugeFileCommand extends ImportFileBaseCommand {
+
+    constructor(client: Client){
+        super(client, 'Huge_Size_File_Test')
+        this.numberOfEntities = 4000;
+    }
+
+    async initData() {
+        const itemsService = new ItemsService(this.papiClient);
+        this.items = await itemsService.prepareUserItems(100000);
```
Comment (simcha-waintrob) [inline server-side/integration-tests/test-commands/import-tests/import-file/huge-file-command.ts:16]: add comment please

File: server-side/integration-tests/services/items-service.ts [hunk 5]
```diff
@@ -1,49 +1,75 @@
...
         }
         return items;
     }
 
+    // resource search return 1000 items per page, we need list of 4000 items for huge test so we need to get 4 pages
     async getUsersItems(): Promise<ItemRelations[]> {
+        let items: ItemRelations[] = [];
+        // for huge test we need the stock to be with 100000 items but the file only contain 4K enteries
+        // for small test we need 500 entieries
+        const itemsToAddCounter: number = this.numberOfItems === 100000 ? 4003 : this.numberOfItems;
```
Comment (simcha-waintrob) [inline server-side/integration-tests/services/items-service.ts:30]: this function should only get items 
talk to Saar to use PageSize -1

**Pull request merged**